feat: [P4PU-195] Status info (transaction status) removal by config

stratoivandiluccio opened the pull request:
<!--- Please always add a PR description as if nobody knows anything about the context these changes come from. -->
<!--- Even if we are all from our internal team, we may not be on the same page. -->
<!--- Write this PR as you were contributing to a public OSS project, where nobody knows you and you have to earn their trust. -->
<!--- This will improve our projects in the long run! Thanks. -->

#### List of Changes

- added a new env variable (SHOW_STATUS_INFO) to hide or show the status info in the whole application
- added a runtime type checking of env variables to ensure compatible values and have a feedback if not (trough zod)
- added unit tests

<!--- Describe your changes in detail -->

#### Motivation and Context
Status info will be introduced after, but we prefer to turn on/off the relative UI instead of commenti/eliminating code
<!--- Why is this change required? What problem does it solve? -->

#### How Has This Been Tested?
Tested locally changing the env variables in the .env file and running/checking the FE subsequently

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->

#### Screenshots (if appropriate):

#### Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

AnisLucidi01 reviewed (approved):
Looks good to me, but I struggle to understand what is causing the problem here: 
<img width="687" alt="image" src="https://github.com/pagopa/pagopa-arc-fe/assets/125553099/6f778149-8d28-4fd0-9c3f-a673df046caf">

stratoivandiluccio commented:
> Looks good to me, but I struggle to understand what is causing the problem here: <img alt="image" width="687" src="https://private-user-images.githubusercontent.com/125553099/339839047-6f778149-8d28-4fd0-9c3f-a673df046caf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2OTk5NDQsIm5iZiI6MTcxODY5OTY0NCwicGF0aCI6Ii8xMjU1NTMwOTkvMzM5ODM5MDQ3LTZmNzc4MTQ5LThkMjgtNGZkMC05YzNmLWE2NzNkZjA0NmNhZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxOFQwODM0MDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05N2VkYmQ1MWNhNjJiZWVjNTkyNTIyMTkwMjYwOWFhNDc1NmFjZjc4MTc4MWI4ZmYxOWZhOGRjNjljOGJiNGNiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Vdq7jvabsy8G5-0ml8jN_g0grJf724hz2cZ6hw2IG_0">

I have introduced some conditions not covered by unit tests. Fixed!
